doc.go

```diff
@@ -0,0 +1,55 @@
+package doc
+
+import (
+	"io"
+	"os/exec"
+)
+
+// JobsList stores a map of the Job's containing the linux process key'd by job id (in memory DB)
+type JobsList map[string]*Job
+
+// Job maintains the exec.Cmd struct (containing the underlying os.Process) and the "owner" for authz
+type Job struct {
+	owner string
+	cmd   *exec.Cmd
+}
+
+// JobOpts wraps the options that can be passed to cgroups for the job
+// details at https://facebookmicrosites.github.io/cgroup2/docs/overview
+type JobOpts struct {
+	CpuWeight int    // `cpu.weight`
+	MemLimit  string // `mem.high`
+	IOLatency string // `io.latency`
+}
+
+// JobStatus is an amalgamation of the useful status information available from the exec.Cmd struct of the job and it's underlying os.Process
+type JobStatus struct {
+	ID       string
+	Owner    string
+	PID      int
+	Running  bool
+	ExitCode int
+}
+
+// Worker defines the libraries API on how to start / stop / query status and get output of a job
+type Worker interface {
```

Comment:
Not sure I understand the motivation for this interface. Are these methods also implemented on JobWorker? If so, why provide an interface?

Reply:
Apologies I did not make this clear. This interface is just for documentation purposes, the JobWorker should implement these functions. The library would not provide this interface as any client using the library should wrap an interface around it's own usage of JobWorker. I've updated now to show JobWorker implementing the API.